clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -21,6 +21,16 @@ using namespace clang;
 using namespace tooling;
 using namespace dependencies;
 
+void ModuleDeps::forEachFileDep(llvm::function_ref<void(StringRef)> Cb) const {
+  SmallString<0> PathBuf;
+  PathBuf.reserve(256);
```

Comment:
Curious what people think of having the caller provide this. If the caller used the same buffer for multiple `ModuleDeps` objects, we could avoid even more allocations. I haven't tested the perf of that scheme.

Reply:
I'll revisit that in another PR.